setup.py

```diff
@@ -30,7 +30,7 @@ def extract_names(filename):
 
 def extract_requirements(filename):
     with open(filename, 'r') as requirements_file:
-        return [x[:-1] for x in requirements_file.readlines()]
+        return [x[:-1].split(' #')[0] for x in requirements_file.readlines()]
```

Comment:
Not required, but you could just tokenize the line and return the first token.  Slightly simpler.

`return [x.split()[0] for x in requirements_file.readlines()]`